client/allocrunner/groupservice_hook.go

```diff
@@ -43,12 +46,20 @@ type groupServiceHookConfig struct {
 }
 
 func newGroupServiceHook(cfg groupServiceHookConfig) *groupServiceHook {
+	var shutdownDelay time.Duration
+	tg := cfg.alloc.Job.LookupTaskGroup(cfg.alloc.TaskGroup)
+
+	if tg != nil && tg.ShutdownDelay != nil {
```

Comment:
If tg is nil things are going to panic a few lines down on L65. Should be safe to skip the check here or something is horribly wrong.
```suggestion
	if tg.ShutdownDelay != nil {
```